From 67908742620f83642ee9688a7af0f249a8d79c85 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 28 Nov 2020 18:24:25 +0000
Subject: [PATCH] GCC: do not pass unique_ptr to DCHECK_NE, but the actual
 pointer

DCHECK_NE comparison requires CheckOpValueStr to be defined for the
type, or providing an output stream operator. A std::unique_ptr
does not provide any.
---
 content/browser/renderer_host/render_frame_host_manager.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/content/browser/renderer_host/render_frame_host_manager.cc b/content/browser/renderer_host/render_frame_host_manager.cc
index 4cb4a1b..54af6d5 100644
--- a/content/browser/renderer_host/render_frame_host_manager.cc
+++ b/content/browser/renderer_host/render_frame_host_manager.cc
@@ -713,7 +713,7 @@ void RenderFrameHostManager::ValidateSpeculativeRenderFrameHostForBug1146573() {
     // This should never be true.
     SCOPED_CRASH_KEY_BOOL(ValidateSpeculative, HostsEqual,
                           speculative_render_frame_host_ == render_frame_host_);
-    DCHECK_NE(speculative_render_frame_host_, render_frame_host_);
+    DCHECK_NE(speculative_render_frame_host_.get(), render_frame_host_.get());
     SCOPED_CRASH_KEY_BOOL(ValidateSpeculative, Live,
                           render_frame_host_->IsRenderFrameLive());
     SCOPED_CRASH_KEY_STRING256(
-- 
2.26.2

